docs/ide-usage.md

```diff
@@ -86,27 +86,118 @@ The following plugins are needed:
+       you downloaded in Step 3 of the [Setting up a development environment](settingUp.md) guide.
+    1. Ensure the `Name:` is of `AppEngine Dev <version>`.
+
+### Automated Setup
+
+If you do not wish to use the automated setup, you can follow the [manual setup](#manual-setup) below.
+
+1. Run this command to create a pre-configured IntelliJ IDEA project:
+   ```sh
+   ./gradlew setupIntellijProject
+   ```
+
+1. Open the project in IntelliJ IDEA. It will generate some user specific settings. Wait for it to finish indexing.
 
-1. In your `Event Log`, you should see this line: `Frameworks detected: Google App Engine, Web, JPA frameworks are detected in the project`. Click `Configure` and `OK` in the dialog box that appears.
+1. Run this command to automatically configure your user specific settings:
+   ```sh
+   ./gradlew setupIntellijProjectSettings
+   ```
+
+1. To set up some static analysis tools, refer to [this document](staticAnalysis.md).
 
-1. Run this command to set up the necessary configuration files for IntelliJ:
+**NOTE**
+>The behavior of the automated setup is described [here](intellij-automated-setup-behavior.md#project-setup-behavior).
+
+### Manual Setup
+
+1. Import the project as a Gradle project as follows:
+   1. If you have no current projects open, click `Import Project`.\
+       **OR**\
+       If you currently have projects open, click `New | Project from Existing Sources...`.
+   1. Select the local repository folder and click `Open`.
+   1. Select `Import project from external model` and then `Gradle`.
+   1. Click `Next`.
+   1. Check `Use auto-import` and uncheck `Create separate module per source set`.
+   1. Ensure `Create directories for empty content root automatically` is unchecked.
+   1. Ensure `Use default gradle wrapper` is selected.
+   1. Ensure for `Gradle JVM:` that a JDK 7 with a name of `1.7` is selected.
+   1. Click `Finish`.
+   1. Create a `build` folder in your project root if it does not exist while waiting for IntelliJ to finish indexing.
+   1. If you used a different project root directory other than `teammates`, you will receive a dialog box with the
+      message `The modules below are not imported from Gradle anymore. Check those to be removed from the ide project too:`.\
+      You should see a module with your project root directory name being checked. Click `OK`.
+   1. Go to `File → Project Structure... → Modules`.
+      Click on the `teammates` module, then under `Sources`, click on the `build` folder and click `Excluded` and then `OK`.
+   1. You should see a dialog box with the message:\
+      `Frameworks detected: Google App Engine, Web, JPA frameworks are detected in the project`.\
+      **OR**\
+      `Frameworks detected: OSGi, Google App Engine, Web, JPA frameworks are detected in the project`.\
+      Click `Configure`, ensure that there is only `Google App Engine` and `JPA` frameworks being shown, otherwise make
```

Comment:
Click `Configure`~~,~~ **and** ensure that ~~there is~~ only `Google App Engine` and `JPA` frameworks ~~being~~ **are** shown,